### PR TITLE
Feature/login interactor

### PR DESCRIPTION
**Pull request by LehlohonoloIsaac**

### CODE REVIEW

File: Showcase-iOS/Login/Interactor/LoginInteractor.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Lehlohonolo Mbele on 2018/04/20.
+//  Copyright © 2018 DVT. All rights reserved.
+//
+
+import Foundation
+
+class LoginInteractor: PresenterInteractable {
+    var loginPresenter: InteractorPresentable?
+    var userAuthenticator: Authenticating?
```
Comment (pjwelcome) [inline Showcase-iOS/Login/Interactor/LoginInteractor.swift:13]: how does this get init'd?

File: Showcase-iOS/Login/Service/UserAuthentication.swift [hunk 2]
```diff
@@ -9,15 +9,11 @@
 import Foundation
 import FirebaseAuth
 
-class UserAuthentication {
-    private var authenticator: FirebaseAuthenticating
+class UserAuthentication: Authenticating {
+    var authenticator: FirebaseAuthenticating?
```
Comment (pjwelcome) [inline Showcase-iOS/Login/Service/UserAuthentication.swift:13]: how does this also get init'd? what your doing doesnt make any sense. I dont really get what this code is doing. seems like you have lots of duplication of code everywhere, with different protocols that do the same thing. Please make sure this code is still applicable.

File: Showcase-iOS/Login/Interactor/LoginInteractor.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    
+    func signIn(withEmail email: String, password: String) {
+        self.userAuthenticator?.signIn(withEmail: email, password: password) { [weak self] (user, error) in
```
Comment (pjwelcome) [inline Showcase-iOS/Login/Interactor/LoginInteractor.swift:16]: where does this get unit tested ?

**Review by pjwelcome — COMMENTED** [review]
Please merge develop into this branch so that you can get the code coverage stats. Thanks.

File: Showcase-iOS/Login/Service/UserAuthentication.swift (hunk 2 above)
Comment (LehlohonoloIsaac) [reply Showcase-iOS/Login/Service/UserAuthentication.swift:13]: I am using property injection to inject them.

File: Showcase-iOS/Login/Interactor/LoginInteractor.swift (hunk 1 above)
Comment (LehlohonoloIsaac) [reply Showcase-iOS/Login/Interactor/LoginInteractor.swift:13]: I am using property injection to inject these.

File: Showcase-iOS/Login/Interactor/LoginInteractor.swift (hunk 3 above)
Comment (LehlohonoloIsaac) [reply Showcase-iOS/Login/Interactor/LoginInteractor.swift:16]: I have already written the tests for the authenticator. Also, this pull request consists of the tests for this LoginInteractor.

**Comment by pjwelcome** [comment]
As we chatted offline, Ill merge this for now, but we need to have a look at why you change this as reasons don't sound like this was necessary.

**Review by pjwelcome — APPROVED** [review]

**Pull request merged**